### PR TITLE
Add Doctrine processors

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Component/Doctrine/Common/State/PersistProcessor.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+final class PersistProcessor implements ProcessorInterface
+{
+    public function __construct(private ManagerRegistry $managerRegistry)
+    {
+    }
+
+    public function process(mixed $data, Operation $operation, Context $context): mixed
+    {
+        $metadata = $context->get(MetadataOption::class)?->metadata();
```
Comment (loic425) [inline src/Component/Doctrine/Common/State/PersistProcessor.php]: I'd like to get rid of the current MetadataInterface and using the new Resource metadata instead.

So I'd like to retrieve a new ResourceOption::class instead.

File: src/Component/Doctrine/Common/Metadata/Resource/Factory/DoctrineResourceMetadataCollectionFactory.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+use Sylius\Component\Resource\Doctrine\Common\State\RemoveProcessor;
+use Sylius\Component\Resource\Metadata\DeleteOperationInterface;
+use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Component\Resource\Metadata\Operations;
+use Sylius\Component\Resource\Metadata\RegistryInterface;
+use Sylius\Component\Resource\Metadata\Resource as ResourceMetadata;
+use Sylius\Component\Resource\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use Sylius\Component\Resource\Metadata\Resource\ResourceMetadataCollection;
+
+class DoctrineResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
```
Comment (lchrusciel) [inline src/Component/Doctrine/Common/Metadata/Resource/Factory/DoctrineResourceMetadataCollectionFactory.php:26]: Why didn't you finalized this class? Is there a proxy created from it?
```suggestion
final class DoctrineResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
```

**Review by lchrusciel — COMMENTED** [review]
We are lacking tests for all new classes from this PR :/

**Review by lchrusciel — APPROVED** [review]

File: src/Component/Doctrine/Common/Metadata/Resource/Factory/DoctrineResourceMetadataCollectionFactory.php (hunk 2 above)
Comment (loic425) [reply src/Component/Doctrine/Common/Metadata/Resource/Factory/DoctrineResourceMetadataCollectionFactory.php:26]: Oh no, that's a mistake

**Pull request merged**